### PR TITLE
Parse error messages, if any, constructing a useful error string to retu...

### PR DESCRIPTION
**Pull request by david-martin**
...rn

### CODE REVIEW

**Comment by david-martin** [comment]
@cianclarke & @jasonmadigan thoughts on error strings being returned here? (see test/errors.js comment for sample messages)

**Comment by jasonmadigan** [comment]
Looks like a good list to me

**Comment by david-martin** [comment]
thanks.
based off the rhc gem https://github.com/openshift/rhc/blob/552a5668e0f9ce1c7a870534aed066124d6e3f9d/lib/rhc/rest/client.rb#L735

**Comment by cianclarke** [comment]
Looks great, Dave! Cherry on top would be a README.md heading on errors with one simple example error-response, but no big deal if no time

File: README.md [hunk 1]
```diff
@@ -124,6 +124,88 @@ This is useful when you need to do a list domains before deciding which one to u
       
     });  
   
+
+### Sample Errors
```
Comment (cianclarke) [inline README.md]: :memo: :+1:

**Pull request merged**